### PR TITLE
Remove pkg/controller/route from golint failures

### PR DESCRIPTION
**Pull request by willdot**
**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
Comments added to the exported types and functions
**Which issue(s) this PR fixes**:

1 part of #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs

```

### CODE REVIEW

**Comment by willdot** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please follow instructions at https://git.k8s.io/community/CLA.md#the-contributor-license-agreement to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.
> 
> * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please log a ticket with the Linux Foundation Helpdesk: https://support.linuxfoundation.org/
> * Should you encounter any issues with the Linux Foundation Helpdesk, send a message to the backup e-mail support address at: [login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)
> 
> Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md). If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).

CLA signed

**Comment by willdot** [comment]
> @willdot: The following tests **failed**, say `/retest` to rerun them all:
> 
> Test name	Commit	Details	Rerun command
> pull-kubernetes-e2e-gce-device-plugin-gpu	[262b9c6](https://github.com/kubernetes/kubernetes/commit/262b9c666e22a02e1b7898b84b9a5014a04a8d47)	[link](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/84003/pull-kubernetes-e2e-gce-device-plugin-gpu/1184512428340154368/)	`/test pull-kubernetes-e2e-gce-device-plugin-gpu`
> pull-kubernetes-e2e-gce-100-performance	[262b9c6](https://github.com/kubernetes/kubernetes/commit/262b9c666e22a02e1b7898b84b9a5014a04a8d47)	[link](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/84003/pull-kubernetes-e2e-gce-100-performance/1184512432530264064/)	`/test pull-kubernetes-e2e-gce-100-performance`
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes&repo=kubernetes&pr=84003). [Your PR dashboard](https://gubernator.k8s.io/pr/willdot). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open) when you hit one in your PR.

/retest

**Comment by willdot** [comment]
> @willdot: The following test **failed**, say `/retest` to rerun them all:
> 
> Test name	Commit	Details	Rerun command
> pull-kubernetes-kubemark-e2e-gce-big	[bf651d1](https://github.com/kubernetes/kubernetes/commit/bf651d164797c02af2eb510d5e68609af2cbfe7c)	[link](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/84003/pull-kubernetes-kubemark-e2e-gce-big/1184537340530397184/)	`/test pull-kubernetes-kubemark-e2e-gce-big`
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes&repo=kubernetes&pr=84003). [Your PR dashboard](https://gubernator.k8s.io/pr/willdot). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open) when you hit one in your PR.

/retest

File: pkg/controller/route/route_controller.go [hunk 1]
```diff
@@ -94,7 +96,8 @@ func New(routes cloudprovider.Routes, kubeClient clientset.Interface, nodeInform
 	return rc
 }
 
-func (rc *RouteController) Run(stopCh <-chan struct{}, syncPeriod time.Duration) {
+// Run starts the route controller until the provided stop channel is closed
```
Comment (cheftako) [inline pkg/controller/route/route_controller.go]: Quibbling but my confidence that the channel will actually stop the controller is not high. (I don't think we've ever actually  tested that it works). Might be better to say something like "starts the controller reconciling routes against node changes every five minutes".

**Comment by cheftako** [comment]
Please resolve the merge conflict

File: hack/.golint_failures [hunk 2]
```diff
@@ -85,7 +85,7 @@ pkg/controller/replication
 pkg/controller/replication/config/v1alpha1
 pkg/controller/resourcequota
 pkg/controller/resourcequota/config/v1alpha1
-pkg/controller/route
+pkg/controller/service
```
Comment (m3ngyang) [inline hack/.golint_failures]: just remove `pkg/controller/route` here.

**Review by cheftako — COMMENTED** [review]
/lgtm

**Pull request closed without merging**